paasta_tools/secret_providers/vault.py

```diff
@@ -159,3 +164,34 @@ def get_secret_signature_from_data(self, data: Mapping[str, Any]) -> Optional[st
             return data["environments"][ecosystem]["signature"]
         else:
             return None
+
+    def get_vault_key_versions(
+        self,
+        key: str,
+        mountpoint: str = "keystore",
+    ) -> Iterable[Dict[str, Any]]:
+        """
+        Retrieve Vault key's all versions based on its metadata
+        """
+        client = self.clients[self.ecosystems[0]]
+        try:
+            meta_response = client.secrets.kv.read_secret_metadata(
+                path=key, mount_point=mountpoint
+            )
+
+            for key_version in meta_response["data"]["versions"].keys():
+                key_response = client.secrets.kv.read_secret_version(
+                    path=key, version=key_version, mount_point=mountpoint
+                )
+                yield {
+                    "keyname": key,
+                    "key": key_response["data"]["data"]["key"],
+                    "key_version": key_response["data"]["metadata"]["version"],
+                }
+        except hvac.exceptions.VaultError as e:
+            log.warning(
+                f"Could not fetch key versions for {key} on {self.ecosystems[0]} because of {type(e).__name__}"
+            )
+            pass
```

Comment:
we can clean this up a little - the python logging system has a helper for automatically including traceback information alongside a log message, and we don't need a `pass` in a block that is otherwise non-empty
```suggestion
        except hvac.exceptions.VaultError:
            log.exception(
                f"Could not fetch key versions for {key} on {self.ecosystems[0]}"
            )
```

Reply:
The `yield` below should capture any cases falling through

Reply:
correct - so we should be able to remove the `pass` :)